[haptics][ios] WIP: Add support for iOS Core Haptics with pattern playback

vovacodes opened the pull request:
# Why

This PR integrates a part of iOS [Core Haptics](https://developer.apple.com/documentation/corehaptics/) API into `expo-haptics` that allows playing custom haptic patterns. It omits the audio-related part of the framework and only focuses on the subset of API that works with haptic events.

We have this functionality implemented as a custom native expo module in [Fuse](https://fusewallet.com) app, we find it quite useful and potentially benefitting the community. Also: https://x.com/tsapeta/status/1992532424086127087 😀

# How

This is just still a WIP, I wanted to check if this approach makes sense for Expo, before writing docs and examples, so would be cool if @tsapeta (or whoever is responsible for this module) could take a look and give a green light to proceed.

# Test Plan

TODO

# Checklist

- [ ] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

vovacodes committed:
[haptics][ios] WIP: Add support for iOS Core Haptics with pattern playback

tsapeta commented on packages/expo-haptics/ios/HapticsModule.swift:
```diff
@@ -66,3 +82,129 @@ public class HapticsModule: Module {
+    hapticEngine?.stop(completionHandler: nil)
+    hapticEngine = nil
+  }
+}
```
Could you move this to `HapticsModule`? We generally try to avoid using extensions for classes/structs within the same package. They are useful for:
- extending external types, i.e. platform and 3rd-party types
- providing default implementation for protocols

The same applies to the extensions for records.

tsapeta commented on packages/expo-haptics/ios/ExpoHaptics.podspec:
```diff
@@ -18,6 +18,7 @@ Pod::Spec.new do |s|
   s.static_framework = true
 
   s.dependency 'ExpoModulesCore'
+  s.frameworks = 'CoreHaptics'
```
This is not necessary, but you can leave it 😉

tsapeta reviewed (commented):
Looks good so far, left a few comments to make it more consistent with the rest of the codebase.

claude committed:
Remove unnecessary s.frameworks declaration

CoreHaptics framework is automatically linked when imported in Swift code.

claude committed:
Address PR review feedback

- Remove unnecessary s.frameworks declaration from podspec
- Move private extension methods into HapticsModule class
- Extract haptic pattern records to separate HapticPattern.swift file
- Rename HapticPatternData to HapticPattern
- Convert string-based type parsing to Swift enums with Enumerable protocol

vovacodes committed:
Merge pull request #1 from vovacodes/claude/expo-haptics-review-fixes-ars13

Claude/expo haptics review fixes ars13
